docs/cpp/source/notes/inference_mode.rst

```diff
@@ -0,0 +1,122 @@
+Inference Mode
+==============
+
+``c10::InferenceMode`` is a new RAII guard analogous to ``NoGradMode``
+to be used when you are certain your operations will have no interactions
+with autograd (e.g. model training). Compared to ``NoGradMode``, code run
+under this mode gets better performance by disabling autograd related work like
+view tracking and version counter bumps. However, tensors created inside
+``c10::InferenceMode`` has more limitation when interacting with autograd system as well.
+
+``InferenceMode`` can be enabled for a given block of code. Inside ``InferenceMode``
+all newly allocated (non-view) tensors are marked as inference tensors. Inference tensors:
+
+- do not have a version counter so an error will be raised if you try to read their version
+  (e.g., because you saved this tensor for backward).
+- are immutable outside ``InferenceMode``. So an error will be raised if you try to:
+  - mutate their data outside InferenceMode.
+  - mutate them into ``requires_grad=True`` outside InferenceMode.
+  To work around you can make a clone to get a normal tensor before mutating.
+
+A non-view tensor is an inference tensor if and only if it was allocated inside ``InferenceMode``.
+A view tensor is an inference tensor if and only if the tensor it is a view of is an inference tensor.
+
+Inside an ``InferenceMode`` block, we make the following performance guarantees:
+
+- Like ``NoGradMode``, all operations do not record ``grad_fn`` even if their ``requires_grad=True``.
+  This applies to both inference tensors and normal tensors.
+- View operations on inference tensors do not do view tracking. View and base inference tensors are
+  indistinguishable.
+- Inplace operations on inference tensors are guaranteed not to do a version bump.
+
+For more implementation details of ``InferenceMode`` please see the `InferenceMode RFC <https://github.com/pytorch/rfcs/pull/17>`_.
```

Comment:
Should the rfc be landed now that it is implemented? Or the goal is to keep them as PRs forever?